lib/improver/utilities/cube_manipulation.py

```diff
@@ -193,12 +188,369 @@ def concatenate_cubes(
+    Warn:
+        Warning: If it does not know what to do with an unmatching
+                 attribute. Default is to delete it.
+    """
+    unmatching_attributes = compare_attributes(cubes)
+    if len(unmatching_attributes) > 0:
+        for i, cube in enumerate(cubes):
+            # Remove history.
+            if "history" in unmatching_attributes[i]:
+                cube.attributes.pop("history")
+                unmatching_attributes[i].pop("history")
+            # Normalise grid_id to ukx_standard_1
+            if "grid_id" in unmatching_attributes[i]:
+                if cube.attributes['grid_id'] in ['enukx_standard_v1',
+                                                  'ukvx_standard_v1']:
+                    cube.attributes['grid_id'] = 'ukx_standard_v1'
+                    unmatching_attributes[i].pop("grid_id")
+            # Add model_id if titles do not match.
+            if "title" in unmatching_attributes[i]:
+                    model_title = cube.attributes.pop('title')
+                    new_model_id_coord = build_coordinate([100*i],
+                                                          long_name='model_id',
+                                                          data_type=np.int)
+                    new_model_coord = build_coordinate([model_title],
+                                                       long_name='model',
+                                                       coord_type=AuxCoord,
+                                                       data_type=np.str)
+                    cube.add_aux_coord(new_model_id_coord)
+                    cube.add_aux_coord(new_model_coord)
+                    unmatching_attributes[i].pop("title")
+            # Remove any other mismatching attributes but raise warning.
+            if len(unmatching_attributes[i]) != 0:
+                for key in unmatching_attributes[i]:
+                    msg = ('Do not know what to do with ' + key +
+                           ' will delete it')
+                    warnings.warn(msg)
+                    cube.attributes.pop(key)
+
+
+def equalise_cube_coords(cubes):
+    """
+    Function to equalise coordinates that do not match.
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to check the coords and revise.
+    Returns:
+        cubelist : Iris cubelist
+            List of cubes with revised coords.
+    Raises:
+        If Percentile coordinates do not match.
+        If Threshold coordinates do not match.
+        If model_id has more than one point.
+    """
+    unmatching_coords = compare_coords(cubes)
+    cubelist = iris.cube.CubeList([])
+    if len(unmatching_coords) > 0:
+        for i, cube in enumerate(cubes):
+            slice_over_keys = []
+            for key in unmatching_coords[i]:
+                # mismatching percentiles - raise error.
+                if key.find('percentile_over') > 0:
+                    msg = ("Percentile coordinates "
```

Comment:
We can only really merge together two percentile cubes if the cube are identical apart from the percentile. In theory we could add in missing percentiles from two mismatching cubes e.g. one model has 0,10,20...,100 the other 0,5,10,20,.....,100 by calculating the missing percentiles using ecc but this starts to get complicated. At the moment I can't see a scenario where we would want to merge together mismatching percentiles as we currently create them all to a set of values at one go, so I have set this up to throw a error if it encounters this. If we find a use case for merging percentiles we can look at how we do it with some solid examples rather than writing code we might never use.

Reply:
As the acceptance criteria says "Handle cubes with realizations from archive and percentile cubes from percentile plugin.", I think that it is fine to take this approach right now.